src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -119332,5 +119332,46 @@ class C<T>
                 //         P6 = t; // 12
                 Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "t").WithLocation(18, 14));
         }
+
+        [Fact]
+        [WorkItem(38586, "https://github.com/dotnet/roslyn/issues/38586")]
+        public void SuppressSwitchExpressionInput()
+        {
+            var source =
+@"#nullable enable
+public class C {
+    public int M0(C a)  => a switch { C _ => 0 }; // ok
+    public int M1(C? a) => a switch { C _ => 0 }; // warns
+    public int M2(C? a) => a! switch { C _ => 0 }; // ok
+
+    public int M3(C a)  => (1, a) switch { (_, C _) => 0 }; // ok
+    public int M4(C? a) => (1, a) switch { (_, C _) => 0 }; // warns
+    public int M5(C? a) => (1, a!) switch { (_, C _) => 0 }; // ok
+
+    public void M6(C? a, bool b)
+    {
+        if (a == null) return;
+        switch (a!)
+        {
+            case C _:
+                break;
+            case null: // does not affect knowledge of 'a'
+                break;
+        }
+
+        a.ToString(); 
```

Comment:
Maybe it would be good to add an `M7` where everything is the same as `M6` except we `switch (a)`, and we *do* get a warning here.

Reply:
No, `switch (a)` does not have exhaustiveness warnings.

---
In reply to: [354012561](https://github.com/dotnet/roslyn/pull/39697#discussion_r354012561) [](ancestors = 354012561)